src/bootstrap/bootstrap.py

```diff
@@ -888,7 +888,7 @@ def build_bootstrap(self, color, verbose_count):
         if target_linker is not None:
             env["RUSTFLAGS"] += " -C linker=" + target_linker
         env["RUSTFLAGS"] += " -Wrust_2018_idioms -Wunused_lifetimes"
-        if self.get_toml("deny-warnings", "rust") != "false":
+        if self.get_toml("deny-warnings", "rust") != "false" and warnings != "warn":
```

Comment:
this doesn't look right - it means `--warnings deny` will be ignored if you have `rust.deny-warnings=false` in config.toml. I think the logic should be something like
```suggestion
        if warnings == "default":
            deny_warnings = self.get_toml("deny-warnings", "rust") != "false"
        else:
            deny_warnings = warnings == "deny"
        if  deny_warnings:
```

Reply:
Ah, I didn't think of precedence. Thanks.